src/v/kafka/server/handlers/topics/types.h

```diff
@@ -58,6 +58,30 @@ static constexpr std::string_view topic_property_data_policy_function_name
 static constexpr std::string_view topic_property_data_policy_script_name
   = "redpanda.datapolicy.script.name";
 
+// Kafka topic properties that is not relevant for Redpanda
+static constexpr std::string_view allowlist_topic_noop_confs[20] = {
```

Comment:
I kind of think that this list should be runtime-updatable as a configuration property.

The reason is that:

1. we might miss something in development / testing and then a customer may discover the capability to update is blocked. we could simply remove the entry from config without requiring a restart or new release.
2. we might want to add something to the list

An example would be `"segment.index.bytes` where we have a corresponding config for this, but apparently it isn't wired up or updatable. Committing that into the binary seems risky.

Reply:
what do you think @jcsp presumably there isn't an issue holding a longish list in centralized config?

Reply:
No problem putting a list this size in there.  We only serialize these when they're actually changed, so it's not expensive (and if user doesn't customize it it'll never be written/read at all).

However, there's a practical risk to making this configurable:
1. Someone edits the list on the field, the modified list gets stored
2. We add support for one of these items, remove it from this _default_ list and release it.
3. User upgrades.  The property doesn't work, because their list of things to ignore is frozen since the time they edited, and changing the default doesn't affect them.

So I think if we make it configurable, then there should still be the hardcoded list of things we definitely don't support, and then the config property should only be for extra items to ignore.  That way, when we update our hardcoded list, it won't conflict with any customization done in the field.

Reply:
Oh great point about edits originating from anywhere and being stored. Presumably this list will change some what in lock step with releases that add/edit support for config options. So needing to change them at runtime is probably less likely than I would imagined.

I'm in support of either approach!